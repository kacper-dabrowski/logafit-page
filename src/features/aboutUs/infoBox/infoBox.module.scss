@import "../../../shared/styles/variables/breakpoints.scss";

.infoBox {
  width: 60%;
  min-width: 300px;
  font-size: 18px;
  line-height: 36px;
  padding: 8px;
  white-space: normal;
  height: 140px;
  margin: auto;

  @media (min-width: $screen-md) {
    width: auto;
  }

  @media (min-width: $screen-lg) {
    padding: 8px 24px;
    font-size: 18px;
  }

  position: relative;
  text-align: center;

  --color-primary: transparent;
  --color-secondary: transparent;
  --gradient-radius: 180deg;

  &.green {
    --color-primary-dark: #2e893e;
    --color-primary: #8ece42;
    --color-secondary: #8fe2d2;
  }

  &.blue {
    --color-primary-dark: #5e53a1;
    --color-primary: #8174d7;
    --color-secondary: #68ceff;
    --gradient-radius: 190deg;
  }

  &.darkblue {
    --color-primary-dark: #37335d;
    --color-primary: #37335d;
    --color-secondary: #3196e2;
    --gradient-radius: 160deg;
  }

  &.lime {
    --color-primary-dark: #83c52c;
    --color-primary: #a0f434;
    --color-secondary: #f4ca34;
    --gradient-radius: 90deg;
  }

  &.skyBlue {
    --color-primary-dark: #37335d;
    --color-primary: #3296e4;
    --color-secondary: #a6e2ff;
    --gradient-radius: 180deg;
  }

  border: 8px solid transparent;
  border-radius: 25px;
  background:
    linear-gradient(#f6fbf3, #f6fbf3) padding-box,
    linear-gradient(
        var(--gradient-radius),
        var(--color-primary) 0%,
        var(--color-secondary) 100%
      )
      border-box;

  .textBlack {
    margin-top: 5px;
    font-weight: bold;
    text-align: left;
    width: 100%;
    display: block;
    font-size: 20px;
  }

  .textColor {
    margin: 5px;
    font-weight: bold;
    color: var(--color-primary-dark);
    font-size: 32px;
    text-align: left;
    word-wrap: break-word;
    display: block;

    @media (min-width: $screen-lg) {
      font-size: 42px;
    }
  }
}
