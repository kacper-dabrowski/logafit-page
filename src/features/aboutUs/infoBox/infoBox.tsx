import clsx from "clsx";
import React from "react";
import styles from "./infoBox.module.scss";

interface InfoBoxProps {
  title: string;
  text: string;
  variant: "green" | "blue" | "lime" | "skyBlue" | "darkblue";
}

export function InfoBox({ title, text, variant }: InfoBoxProps) {
  const classes = clsx(styles.infoBox, styles[variant]);

  return (
    <div className={classes}>
      <div>
        <span className={styles.textBlack}>{title}</span>
        <span className={styles.textColor}>{text}</span>
      </div>
    </div>
  );
}
