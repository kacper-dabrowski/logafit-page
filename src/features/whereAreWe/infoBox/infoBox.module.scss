@import "../../../shared/styles/variables/breakpoints.scss";

.infoBox {
  width: auto;
  font-size: 18px;
  line-height: 36px;
  padding: 0;
  white-space: nowrap;

  @media (min-width: $screen-lg) {
    padding: 8px 24px;
    font-size: 18px;
  }

  position: relative;
  text-align: center;

  --color-primary: transparent;
  --color-secondary: transparent;
  --gradient-radius: 180deg;

  &.green {
    --color-primary-dark: #2e893e;
    --color-primary: #8ece42;
    --color-secondary: #8fe2d2;
  }

  &.blue {
    --color-primary-dark: #5e53a1;
    --color-primary: #8174d7;
    --color-secondary: #68ceff;
    --gradient-radius: 90deg;
  }

  &.lime {
    --color-primary-dark: #83c52c;
    --color-primary: #a0f434;
    --color-secondary: #f4ca34;
    --gradient-radius: 90deg;
  }

  &.skyBlue {
    --color-primary-dark: #37335d;
    --color-primary: #3296e4;
    --color-secondary: #a6e2ff;
    --gradient-radius: 180deg;
  }

  border: 8px solid transparent;
  border-radius: 25px;
  background:
    linear-gradient(#dfedf5, #dfedf5) padding-box,
    linear-gradient(
        var(--gradient-radius),
        var(--color-primary) 0%,
        var(--color-secondary) 100%
      )
      border-box;

  .textBlack {
    margin: 5px;
    font-weight: bold;
  }

  .textColor {
    margin: 5px;
    font-weight: bold;
    color: var(--color-primary-dark);
    font-size: 28px;
  }
}
