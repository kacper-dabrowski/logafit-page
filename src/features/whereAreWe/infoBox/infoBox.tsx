import clsx from "clsx";
import React from "react";
import styles from "./infoBox.module.scss";

interface InfoBoxProps {
  text: string;
  variant: "green" | "blue" | "lime" | "skyBlue";
  separator?: string;
}

export function InfoBox({ text, variant, separator }: InfoBoxProps) {
  const classes = clsx(styles.infoBox, styles[variant]);
  let textA;
  if (separator) {
    textA = text.split(separator);
  } else {
    textA = [text];
  }
  return (
    <div className={classes}>
      <div>
        {textA.map((content, index) => (
          <>
            <span className={styles.textColor}>{content}</span>
            {index < textA.length - 1 && <span>{separator}</span>}
          </>
        ))}
      </div>
    </div>
  );
}
