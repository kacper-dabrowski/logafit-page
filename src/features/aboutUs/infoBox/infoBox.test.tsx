import { render, screen } from "@testing-library/react";
import { InfoBox } from "./infoBox";

describe("infoBox section", () => {
  it("displays infoBox title and text correcly ", () => {
    render(
      <InfoBox title="Telefonicznie" text="607 880 086" variant="green" />,
    );

    expect(screen.getByText("Telefonicznie")).toBeInTheDocument();
    expect(screen.getByText("607 880 086")).toBeInTheDocument();
  });

  it("infoBox has the correct structure", () => {
    const { container } = render(
      <InfoBox title="Telefonicznie" text="607 880 086" variant="green" />,
    );

    expect(container).toMatchSnapshot();
  });
});
