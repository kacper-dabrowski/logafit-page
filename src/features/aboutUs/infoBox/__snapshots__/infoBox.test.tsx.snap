// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`infoBox section infoBox has the correct structure 1`] = `
<div>
  <div
    class="infoBox green"
  >
    <div>
      <span
        class="textBlack"
      >
        Telefonicznie
      </span>
      <span
        class="textColor"
      >
        607 880 086
      </span>
    </div>
  </div>
</div>
`;
